### PR TITLE
Gabriel/op 243 ticket schema and attestation

### PR DESCRIPTION
**Pull request by gabrielorleans**
Screen Shot 2024-09-11 at 11 13 01

Needs to be placed in the right page in future steps.

Already checked on etherscan https://sepolia-optimism.etherscan.io/tx/0x6417a30244895e8aabe2c7013a4313029f3f94027a02fb7777440fefa0843fdd

### CODE REVIEW

File: src/components/CreateTicketSchemaButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useRegistrySchema } from "@/hooks/useRegistrySchema";
+import { encodePacked, keccak256, zeroAddress } from "viem";
+// import { CREATE_EVENT_SCHEMA } from "./CreateEventSchemaButton";
```
Comment (luizakp) [inline src/components/CreateTicketSchemaButton.tsx]: please remove comments and console logs

File: src/components/CreateTicketSchemaButton.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export const CREATE_TICKET_SCHEMA =
+  "address owner, string eventid";
+
+  export const CREATE_TICKET_SCHEMA_UID = keccak256(
+    encodePacked(
+      ["string", "address", "bool"],
+      [CREATE_TICKET_SCHEMA, zeroAddress, true]
+    )
+  );
+  console.log(CREATE_TICKET_SCHEMA_UID)
```
Comment (luizakp) [inline src/components/CreateTicketSchemaButton.tsx]: https://github.com/bleu/op-bacchus/pull/18/files#r1754856580

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,7 +1,13 @@
+"use client"
+import { CreateTicketAttestationButton } from "@/components/CreateTicketAttestationButton";
+import { CreateTicketSchemaButton } from "@/components/CreateTicketSchemaButton";
+
 export default function Home() {
   return (
     <main className="flex min-h-screen flex-col items-center justify-between p-24">
       <h1>Bacchus Onboarding</h1>
+      <CreateTicketSchemaButton />
+      <CreateTicketAttestationButton />
```
Comment (luizakp) [inline src/app/page.tsx]: this can be added on the` /admin` if you rebase with main

File: src/components/CreateTicketAttestationButton.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { useCreateTicketAttestation } from "../hooks/useCreateTicketAttestation";
+import type { Ticket } from "../hooks/useCreateTicketAttestation";
+import { toast } from "react-toastify";
+
+
+
+const newTicket = {
+  owner: "0xa90914762709441d557De208bAcE1edB1A3968b2",
+  eventid: "test",
+} as Ticket;
```
Comment (luizakp) [inline src/components/CreateTicketAttestationButton.tsx]: appreciate this being a start and a ticket is really being created! but with the current state of the project, you can already create a ticket linked to a real event and with the current connected wallet address

File: src/components/CreateTicketAttestationButton.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    await createTicketAttestation({ ticket: newTicket }).catch((error) => {
+      toast(error, {
+        position: "top-center",
+        type: "warning",
+        autoClose: 3000,
+        theme: "light",
+        pauseOnHover: true,
+      });
+    });
+  }
```
Comment (luizakp) [inline src/components/CreateTicketAttestationButton.tsx]: 🔝

**Review by jean-neiverth — CHANGES_REQUESTED** [review]
It would be better if you rebase / merge main before we can merge the PR, this way (i) you will be able to put the buttons in /admin page and (ii) the reviewers will know all the changes being proposed considering the current main, not the old one

**Pull request merged**